src/frontends/onnx/frontend/CMakeLists.txt

```diff
@@ -16,7 +16,7 @@ ov_add_frontend(NAME onnx
                 LINK_LIBRARIES ngraph::builder onnx_common openvino::core::dev)
 
 set(ONNX_OPSET_VERSION 17 CACHE INTERNAL "Supported version of ONNX operator set")
-target_compile_definitions(${TARGET_NAME} PRIVATE ONNX_OPSET_VERSION=${ONNX_OPSET_VERSION})
+target_compile_definitions(${TARGET_NAME} PUBLIC ONNX_OPSET_VERSION=${ONNX_OPSET_VERSION})
```

Comment:
is this information needed for end users?

Reply:
In the current approach, I'm using it only in tests, but in general I believe that it can be useful for a user (but I haven't a strong opinion here)

Reply:
If we don't have a known use cases why it is needed for the end users, I propose to use PRIVATE definition.

Reply:
in tests we can explicitly add `ONNX_OPSET_VERSION` definition